DQMOffline/RecoB/plugins/MiniAODTaggerAnalyzer.cc

```diff
@@ -0,0 +1,84 @@
+#include "DQMOffline/RecoB/plugins/MiniAODTaggerAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+
+MiniAODTaggerAnalyzer::MiniAODTaggerAnalyzer(const edm::ParameterSet& pSet)
+    : jetToken_(consumes<std::vector<pat::Jet> >(pSet.getParameter<edm::InputTag>("JetTag"))),
+      disrParameters_(pSet.getParameter<edm::ParameterSet>("parameters")),
+
+      folder_(pSet.getParameter<std::string>("folder")),
+      discrNumerator_(pSet.getParameter<vstring>("numerator")),
+      discrDenominator_(pSet.getParameter<vstring>("denominator")),
+
+      mclevel_(pSet.getParameter<int>("MClevel")),
+      doCTagPlots_(pSet.getParameter<bool>("CTagPlots")),
+      dodifferentialPlots_(pSet.getParameter<bool>("differentialPlots")),
+      discrCut_(pSet.getParameter<double>("discrCut")),
+
+      etaActive_(pSet.getParameter<bool>("etaActive")),
+      etaMin_(pSet.getParameter<double>("etaMin")),
+      etaMax_(pSet.getParameter<double>("etaMax")),
+      ptActive_(pSet.getParameter<bool>("ptActive")),
+      ptMin_(pSet.getParameter<double>("ptMin")),
+      ptMax_(pSet.getParameter<double>("ptMax"))
+
+{}
+
+MiniAODTaggerAnalyzer::~MiniAODTaggerAnalyzer() {}
+
+void MiniAODTaggerAnalyzer::bookHistograms(DQMStore::IBooker& ibook, edm::Run const& run, edm::EventSetup const& es) {
+  jetTagPlotter_ = std::make_unique<JetTagPlotter>(folder_,
+                                                   EtaPtBin(etaActive_, etaMin_, etaMax_, ptActive_, ptMin_, ptMax_),
+                                                   disrParameters_,
+                                                   mclevel_,
+                                                   false,
+                                                   ibook,
+                                                   doCTagPlots_,
+                                                   dodifferentialPlots_,
+                                                   discrCut_);
+}
+
+void MiniAODTaggerAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<std::vector<pat::Jet> > jetCollection;
+  iEvent.getByToken(jetToken_, jetCollection);
+
+  const float jec = 1.;  // JEC not implemented!
+  float numerator = 0;
```

Comment:
No need to (define and) initialize `numerator` and `denominator` before the for loop